# Import library

In [1]:
import pandas as pd
import numpy as np

raw_data=pd.read_csv('flight_data.csv')
raw_data

year  month  day  dep_time  sched_dep_time  dep_delay  arr_time  \
0       2013      1    1     517.0             515        2.0     830.0   
1       2013      1    1     533.0             529        4.0     850.0   
2       2013      1    1     542.0             540        2.0     923.0   
3       2013      1    1     544.0             545       -1.0    1004.0   
4       2013      1    1     554.0             600       -6.0     812.0   
5       2013      1    1     554.0             558       -4.0     740.0   
6       2013      1    1     555.0             600       -5.0     913.0   
7       2013      1    1     557.0             600       -3.0     709.0   
8       2013      1    1     557.0             600       -3.0     838.0   
9       2013      1    1     558.0             600       -2.0     753.0   
10      2013      1    1     558.0             600       -2.0     849.0   
11      2013      1    1     558.0             600       -2.0     853.0   
12      2013      1    1     558.0             600       -2.0     924.0   
13      2013      1    1     558.0             600       -2.0     923.0   
14      2013      1    1     559.0             600       -1.0     941.0   
15      2013      1    1     559.0             559        0.0     702.0   
16      2013      1    1     559.0             600       -1.0     854.0   
17      2013      1    1     600.0             600        0.0     851.0   
18      2013      1    1     600.0             600        0.0     837.0   
19      2013      1    1     601.0             600        1.0     844.0   
20      2013      1    1     602.0             610       -8.0     812.0   
21      2013      1    1     602.0             605       -3.0     821.0   
22      2013      1    1     606.0             610       -4.0     858.0   
23      2013      1    1     606.0             610       -4.0     837.0   
24      2013      1    1     607.0             607        0.0     858.0   
25      2013      1    1     608.0             600        8.0     807.0   
26      2013      1    1     611.0             600       11.0     945.0   
27      2013      1    1     613.0             610        3.0     925.0   
28      2013      1    1     615.0             615        0.0    1039.0   
29      2013      1    1     615.0             615        0.0     833.0   
...      ...    ...  ...       ...             ...        ...       ...   
336746  2013      9   30    2123.0            2125       -2.0    2223.0   
336747  2013      9   30    2127.0            2129       -2.0    2314.0   
336748  2013      9   30    2128.0            2130       -2.0    2328.0   
336749  2013      9   30    2129.0            2059       30.0    2230.0   
336750  2013      9   30    2131.0            2140       -9.0    2225.0   
336751  2013      9   30    2140.0            2140        0.0      10.0   
336752  2013      9   30    2142.0            2129       13.0    2250.0   
336753  2013      9   30    2145.0            2145        0.0     115.0   
336754  2013      9   30    2147.0            2137       10.0      30.0   
336755  2013      9   30    2149.0            2156       -7.0    2245.0   
336756  2013      9   30    2150.0            2159       -9.0    2250.0   
336757  2013      9   30    2159.0            1845      194.0    2344.0   
336758  2013      9   30    2203.0            2205       -2.0    2339.0   
336759  2013      9   30    2207.0            2140       27.0    2257.0   
336760  2013      9   30    2211.0            2059       72.0    2339.0   
336761  2013      9   30    2231.0            2245      -14.0    2335.0   
336762  2013      9   30    2233.0            2113       80.0     112.0   
336763  2013      9   30    2235.0            2001      154.0      59.0   
336764  2013      9   30    2237.0            2245       -8.0    2345.0   
336765  2013      9   30    2240.0            2245       -5.0    2334.0   
336766  2013      9   30    2240.0            2250      -10.0    2347.0   
336767  2013      9   30    2241.

# The CSV file contains 19 columns and 336776 records

In [2]:
raw_data.shape

(336776, 19)

In [3]:
raw_data['origin']

0         EWR
1         LGA
2         JFK
3         JFK
4         LGA
5         EWR
6         EWR
7         LGA
8         JFK
9         LGA
10        JFK
11        JFK
12        JFK
13        EWR
14        LGA
15        JFK
16        EWR
17        LGA
18        LGA
19        EWR
20        LGA
21        LGA
22        EWR
23        JFK
24        EWR
25        EWR
26        JFK
27        JFK
28        JFK
29        EWR
         ... 
336746    LGA
336747    EWR
336748    JFK
336749    LGA
336750    JFK
336751    JFK
336752    EWR
336753    JFK
336754    LGA
336755    EWR
336756    EWR
336757    JFK
336758    LGA
336759    LGA
336760    EWR
336761    JFK
336762    EWR
336763    JFK
336764    JFK
336765    JFK
336766    JFK
336767    JFK
336768    JFK
336769    JFK
336770    LGA
336771    JFK
336772    LGA
336773    LGA
336774    LGA
336775    LGA
Name: origin, dtype: object

In [4]:
raw_data.axes

[RangeIndex(start=0, stop=336776, step=1),
 Index([u'year', u'month', u'day', u'dep_time', u'sched_dep_time', u'dep_delay',
        u'arr_time', u'sched_arr_time', u'arr_delay', u'carrier', u'flight',
        u'tailnum', u'origin', u'dest', u'air_time', u'distance', u'hour',
        u'minute', u'time_hour'],
       dtype='object')]

# The data is based on 3 airports in New York that flies over

In [5]:
raw_data.origin.unique()

array(['EWR', 'LGA', 'JFK'], dtype=object)

In [6]:
raw_data.dest.unique()

array(['IAH', 'MIA', 'BQN', 'ATL', 'ORD', 'FLL', 'IAD', 'MCO', 'PBI',
       'TPA', 'LAX', 'SFO', 'DFW', 'BOS', 'LAS', 'MSP', 'DTW', 'RSW',
       'SJU', 'PHX', 'BWI', 'CLT', 'BUF', 'DEN', 'SNA', 'MSY', 'SLC',
       'XNA', 'MKE', 'SEA', 'ROC', 'SYR', 'SRQ', 'RDU', 'CMH', 'JAX',
       'CHS', 'MEM', 'PIT', 'SAN', 'DCA', 'CLE', 'STL', 'MYR', 'JAC',
       'MDW', 'HNL', 'BNA', 'AUS', 'BTV', 'PHL', 'STT', 'EGE', 'AVL',
       'PWM', 'IND', 'SAV', 'CAK', 'HOU', 'LGB', 'DAY', 'ALB', 'BDL',
       'MHT', 'MSN', 'GSO', 'CVG', 'BUR', 'RIC', 'GSP', 'GRR', 'MCI',
       'ORF', 'SAT', 'SDF', 'PDX', 'SJC', 'OMA', 'CRW', 'OAK', 'SMF',
       'TUL', 'TYS', 'OKC', 'PVD', 'DSM', 'PSE', 'BHM', 'CAE', 'HDN',
       'BZN', 'MTJ', 'EYW', 'PSP', 'ACK', 'BGR', 'ABQ', 'ILM', 'MVY',
       'SBN', 'LEX', 'CHO', 'TVC', 'ANC', 'LGA'], dtype=object)

# 105 destinations

In [7]:
len(raw_data.dest.unique())

105

# The top 5 flying destinations are ORD, ATL, LAX, BOS, MCO which make up about 24% of total flights out of the three airports.

In [8]:
grp_df=pd.DataFrame(pd.value_counts(raw_data.dest))
grp_df.columns=['Flights']
grp_df['Percent']=grp_df[0:]/grp_df.sum()*100
grp_df

Flights   Percent
ORD    17283  5.131898
ATL    17215  5.111706
LAX    16174  4.802599
BOS    15508  4.604841
MCO    14082  4.181414
CLT    14064  4.176070
SFO    13331  3.958417
FLL    12055  3.579531
MIA    11728  3.482433
DCA     9705  2.881737
DTW     9384  2.786422
DFW     8738  2.594603
RDU     8163  2.423866
TPA     7466  2.216904
DEN     7266  2.157517
IAH     7198  2.137326
MSP     7185  2.133466
PBI     6554  1.946101
BNA     6333  1.880478
LAS     5997  1.780709
SJU     5819  1.727855
IAD     5700  1.692520
BUF     4681  1.389945
PHX     4656  1.382521
CLE     4573  1.357876
STL     4339  1.288393
MDW     4113  1.221287
CVG     3941  1.170214
SEA     3923  1.164869
MSY     3799  1.128050
..       ...       ...
BGR      375  0.111350
BUR      371  0.110162
PSE      365  0.108381
OKC      346  0.102739
SJC      329  0.097691
TUL      315  0.093534
OAK      312  0.092643
BHM      297  0.088189
SMF      284  0.084329
AVL      275  0.081657
ACK      265  0.078687
ABQ      254  0.075421
MVY      221  0.065622
EGE      213  0.063247
CRW      138  0.040977
CAE      116  0.034444
ILM      110  0.032663
TVC      101  0.029990
MYR       59  0.017519
CHO       52  0.015441
BZN       36  0.010690
JAC       25  0.007423
PSP       19  0.005642
EYW       17  0.005048
MTJ       15  0.004454
HDN       15  0.004454
SBN       10  0.002969
ANC        8  0.002375
LEX        1  0.000297
LGA        1  0.000297

[105 rows x 2 columns]

# Flights by Destination in Percent

In [9]:
grp_df[0:]/grp_df.sum()*100

Flights   Percent
ORD  5.131898  5.131898
ATL  5.111706  5.111706
LAX  4.802599  4.802599
BOS  4.604841  4.604841
MCO  4.181414  4.181414
CLT  4.176070  4.176070
SFO  3.958417  3.958417
FLL  3.579531  3.579531
MIA  3.482433  3.482433
DCA  2.881737  2.881737
DTW  2.786422  2.786422
DFW  2.594603  2.594603
RDU  2.423866  2.423866
TPA  2.216904  2.216904
DEN  2.157517  2.157517
IAH  2.137326  2.137326
MSP  2.133466  2.133466
PBI  1.946101  1.946101
BNA  1.880478  1.880478
LAS  1.780709  1.780709
SJU  1.727855  1.727855
IAD  1.692520  1.692520
BUF  1.389945  1.389945
PHX  1.382521  1.382521
CLE  1.357876  1.357876
STL  1.288393  1.288393
MDW  1.221287  1.221287
CVG  1.170214  1.170214
SEA  1.164869  1.164869
MSY  1.128050  1.128050
..        ...       ...
BGR  0.111350  0.111350
BUR  0.110162  0.110162
PSE  0.108381  0.108381
OKC  0.102739  0.102739
SJC  0.097691  0.097691
TUL  0.093534  0.093534
OAK  0.092643  0.092643
BHM  0.088189  0.088189
SMF  0.084329  0.084329
AVL  0.081657  0.081657
ACK  0.078687  0.078687
ABQ  0.075421  0.075421
MVY  0.065622  0.065622
EGE  0.063247  0.063247
CRW  0.040977  0.040977
CAE  0.034444  0.034444
ILM  0.032663  0.032663
TVC  0.029990  0.029990
MYR  0.017519  0.017519
CHO  0.015441  0.015441
BZN  0.010690  0.010690
JAC  0.007423  0.007423
PSP  0.005642  0.005642
EYW  0.005048  0.005048
MTJ  0.004454  0.004454
HDN  0.004454  0.004454
SBN  0.002969  0.002969
ANC  0.002375  0.002375
LEX  0.000297  0.000297
LGA  0.000297  0.000297

[105 rows x 2 columns]

In [10]:
raw_data.head(10)

year  month  day  dep_time  sched_dep_time  dep_delay  arr_time  \
0  2013      1    1     517.0             515        2.0     830.0   
1  2013      1    1     533.0             529        4.0     850.0   
2  2013      1    1     542.0             540        2.0     923.0   
3  2013      1    1     544.0             545       -1.0    1004.0   
4  2013      1    1     554.0             600       -6.0     812.0   
5  2013      1    1     554.0             558       -4.0     740.0   
6  2013      1    1     555.0             600       -5.0     913.0   
7  2013      1    1     557.0             600       -3.0     709.0   
8  2013      1    1     557.0             600       -3.0     838.0   
9  2013      1    1     558.0             600       -2.0     753.0   

   sched_arr_time  arr_delay carrier  flight tailnum origin dest  air_time  \
0             819       11.0      UA    1545  N14228    EWR  IAH     227.0   
1             830       20.0      UA    1714  N24211    LGA  IAH     227.0   
2             850       33.0      AA    1141  N619AA    JFK  MIA     160.0   
3            1022      -18.0      B6     725  N804JB    JFK  BQN     183.0   
4             837      -25.0      DL     461  N668DN    LGA  ATL     116.0   
5             728       12.0      UA    1696  N39463    EWR  ORD     150.0   
6             854       19.0      B6     507  N516JB    EWR  FLL     158.0   
7             723      -14.0      EV    5708  N829AS    LGA  IAD      53.0   
8             846       -8.0      B6      79  N593JB    JFK  MCO     140.0   
9             745        8.0      AA     301  N3ALAA    LGA  ORD     138.0   

   distance  hour  minute         time_hour  
0      1400     5      15  01-01-2013 05:00  
1      1416     5      29  01-01-2013 05:00  
2      1089     5      40  01-01-2013 05:00  
3      1576     5      45  01-01-2013 05:00  
4       762     6       0  01-01-2013 06:00  
5       719     5      58  01-01-2013 05:00  
6      1065     6       0  01-01-2013 06:00  
7       229     6       0  01-01-2013 06:00  
8       944     6       0  01-01-2013 06:00  
9       733     6       0  01-01-2013 06:00

# EWR airport is the most busy airport with 120835 flights follow by JFK 111279 and LGA 104662 flights 

In [11]:
pd.value_counts(raw_data.origin)

EWR    120835
JFK    111279
LGA    104662
Name: origin, dtype: int64

In [12]:
gg=raw_data[raw_data['origin']=='EWR']
xx=gg.groupby(['origin','dest']).size().reset_index()
xx.columns=['origin','dest','flights']
xx
xx.nlargest(5,'flights')

origin dest  flights
55    EWR  ORD     6100
7     EWR  BOS     5327
74    EWR  SFO     5127
16    EWR  CLT     5026
2     EWR  ATL     5022

In [13]:
yy=pd.DataFrame(raw_data[raw_data['origin']=='EWR']['month'])
yy.groupby('month').size().describe()

count       12.000000
mean     10069.583333
std        452.267109
min       9107.000000
25%       9846.500000
50%      10139.500000
75%      10433.750000
max      10592.000000
dtype: float64

In [14]:
yy=pd.DataFrame(raw_data[raw_data['origin']=='JFK']['month'])
yy.groupby('month').size().describe()

count       12.000000
mean      9273.250000
std        479.208462
min       8421.000000
25%       9084.250000
50%       9189.500000
75%       9528.250000
max      10023.000000
dtype: float64

In [15]:
yy=pd.DataFrame(raw_data[raw_data['origin']=='LGA']['month'])
yy.groupby('month').size().describe()

count      12.000000
mean     8721.833333
std       570.078756
min      7423.000000
25%      8592.250000
50%      8829.000000
75%      9005.500000
max      9642.000000
dtype: float64

# Base on a mean of 10069, EWR is most busy in the month of Mar to Aug
# Base on a mean of 9273, JFK is most busy in the month of Mar, May to Aug
# Base on a mean of 8721, LGA is most busy in the month of May, Jul to Oct and Dec where extra manpower may have to be deployed in the respective airports for smooth operation. 

In [16]:
raw_data.groupby(['origin','month']).size().unstack().reset_index()
#zz=yy[yy['origin']=='EWR']['month']
#zz.describe()

month origin     1     2      3      4      5      6      7      8     9  \
0        EWR  9893  9107  10420  10531  10592  10175  10475  10359  9550   
1        JFK  9161  8421   9697   9218   9397   9472  10023   9983  8908   
2        LGA  7950  7423   8717   8581   8807   8596   8927   8985  9116   

month     10    11    12  
0      10104  9707  9922  
1       9143  8710  9146  
2       9642  8851  9067

# Aircraft Speed Analysis. Most planes are travelling at the mean speed of 6.57122 miles/min

No of NA inside each columns of the dataframe

In [158]:
speed_df=raw_data
speed_df.isnull().sum()

year                 0
month                0
day                  0
dep_time          8255
sched_dep_time       0
dep_delay         8255
arr_time          8713
sched_arr_time       0
arr_delay         9430
carrier              0
flight               0
tailnum           2512
origin               0
dest                 0
air_time          9430
distance             0
hour                 0
minute               0
time_hour            0
speed             9430
dtype: int64

create another column call speed. Taking the speed divide by the airtime for mile/min

In [135]:
speed_df['speed']=speed_df.distance/speed_df.air_time
speed_df

year  month  day  dep_time  sched_dep_time  dep_delay  arr_time  \
0       2013      1    1     517.0             515        2.0     830.0   
1       2013      1    1     533.0             529        4.0     850.0   
2       2013      1    1     542.0             540        2.0     923.0   
3       2013      1    1     544.0             545       -1.0    1004.0   
4       2013      1    1     554.0             600       -6.0     812.0   
5       2013      1    1     554.0             558       -4.0     740.0   
6       2013      1    1     555.0             600       -5.0     913.0   
7       2013      1    1     557.0             600       -3.0     709.0   
8       2013      1    1     557.0             600       -3.0     838.0   
9       2013      1    1     558.0             600       -2.0     753.0   
10      2013      1    1     558.0             600       -2.0     849.0   
11      2013      1    1     558.0             600       -2.0     853.0   
12      2013      1    1     558.0             600       -2.0     924.0   
13      2013      1    1     558.0             600       -2.0     923.0   
14      2013      1    1     559.0             600       -1.0     941.0   
15      2013      1    1     559.0             559        0.0     702.0   
16      2013      1    1     559.0             600       -1.0     854.0   
17      2013      1    1     600.0             600        0.0     851.0   
18      2013      1    1     600.0             600        0.0     837.0   
19      2013      1    1     601.0             600        1.0     844.0   
20      2013      1    1     602.0             610       -8.0     812.0   
21      2013      1    1     602.0             605       -3.0     821.0   
22      2013      1    1     606.0             610       -4.0     858.0   
23      2013      1    1     606.0             610       -4.0     837.0   
24      2013      1    1     607.0             607        0.0     858.0   
25      2013      1    1     608.0             600        8.0     807.0   
26      2013      1    1     611.0             600       11.0     945.0   
27      2013      1    1     613.0             610        3.0     925.0   
28      2013      1    1     615.0             615        0.0    1039.0   
29      2013      1    1     615.0             615        0.0     833.0   
...      ...    ...  ...       ...             ...        ...       ...   
336746  2013      9   30    2123.0            2125       -2.0    2223.0   
336747  2013      9   30    2127.0            2129       -2.0    2314.0   
336748  2013      9   30    2128.0            2130       -2.0    2328.0   
336749  2013      9   30    2129.0            2059       30.0    2230.0   
336750  2013      9   30    2131.0            2140       -9.0    2225.0   
336751  2013      9   30    2140.0            2140        0.0      10.0   
336752  2013      9   30    2142.0            2129       13.0    2250.0   
336753  2013      9   30    2145.0            2145        0.0     115.0   
336754  2013      9   30    2147.0            2137       10.0      30.0   
336755  2013      9   30    2149.0            2156       -7.0    2245.0   
336756  2013      9   30    2150.0            2159       -9.0    2250.0   
336757  2013      9   30    2159.0            1845      194.0    2344.0   
336758  2013      9   30    2203.0            2205       -2.0    2339.0   
336759  2013      9   30    2207.0            2140       27.0    2257.0   
336760  2013      9   30    2211.0            2059       72.0    2339.0   
336761  2013      9   30    2231.0            2245      -14.0    2335.0   
336762  2013      9   30    2233.0            2113       80.0     112.0   
336763  2013      9   30    2235.0            2001      154.0      59.0   
336764  2013      9   30    2237.0            2245       -8.0    2345.0   
336765  2013      9   30    2240.0            2245       -5.0    2334.0   
336766  2013      9   30    2240.0            2250      -10.0    2347.0   
336767  2013      9   30    2241.

In [136]:
speed_df['speed'].describe()

count    327346.000000
mean          6.571228
std           1.010550
min           1.280000
25%           5.968153
50%           6.735849
75%           7.313725
max          11.723077
Name: speed, dtype: float64

drop all rows that contain NAN under speed column

In [142]:
speed_df=speed_df.dropna(subset=['speed'])
speed_df

year  month  day  dep_time  sched_dep_time  dep_delay  arr_time  \
0       2013      1    1     517.0             515        2.0     830.0   
1       2013      1    1     533.0             529        4.0     850.0   
2       2013      1    1     542.0             540        2.0     923.0   
3       2013      1    1     544.0             545       -1.0    1004.0   
4       2013      1    1     554.0             600       -6.0     812.0   
5       2013      1    1     554.0             558       -4.0     740.0   
6       2013      1    1     555.0             600       -5.0     913.0   
7       2013      1    1     557.0             600       -3.0     709.0   
8       2013      1    1     557.0             600       -3.0     838.0   
9       2013      1    1     558.0             600       -2.0     753.0   
10      2013      1    1     558.0             600       -2.0     849.0   
11      2013      1    1     558.0             600       -2.0     853.0   
12      2013      1    1     558.0             600       -2.0     924.0   
13      2013      1    1     558.0             600       -2.0     923.0   
14      2013      1    1     559.0             600       -1.0     941.0   
15      2013      1    1     559.0             559        0.0     702.0   
16      2013      1    1     559.0             600       -1.0     854.0   
17      2013      1    1     600.0             600        0.0     851.0   
18      2013      1    1     600.0             600        0.0     837.0   
19      2013      1    1     601.0             600        1.0     844.0   
20      2013      1    1     602.0             610       -8.0     812.0   
21      2013      1    1     602.0             605       -3.0     821.0   
22      2013      1    1     606.0             610       -4.0     858.0   
23      2013      1    1     606.0             610       -4.0     837.0   
24      2013      1    1     607.0             607        0.0     858.0   
25      2013      1    1     608.0             600        8.0     807.0   
26      2013      1    1     611.0             600       11.0     945.0   
27      2013      1    1     613.0             610        3.0     925.0   
28      2013      1    1     615.0             615        0.0    1039.0   
29      2013      1    1     615.0             615        0.0     833.0   
...      ...    ...  ...       ...             ...        ...       ...   
336740  2013      9   30    2115.0            2110        5.0    2246.0   
336741  2013      9   30    2116.0            2124       -8.0    2317.0   
336742  2013      9   30    2119.0            2005       74.0    2310.0   
336743  2013      9   30    2119.0            2120       -1.0    2330.0   
336744  2013      9   30    2121.0            2100       21.0    2349.0   
336745  2013      9   30    2122.0            2127       -5.0    2226.0   
336746  2013      9   30    2123.0            2125       -2.0    2223.0   
336747  2013      9   30    2127.0            2129       -2.0    2314.0   
336748  2013      9   30    2128.0            2130       -2.0    2328.0   
336749  2013      9   30    2129.0            2059       30.0    2230.0   
336750  2013      9   30    2131.0            2140       -9.0    2225.0   
336751  2013      9   30    2140.0            2140        0.0      10.0   
336752  2013      9   30    2142.0            2129       13.0    2250.0   
336753  2013      9   30    2145.0            2145        0.0     115.0   
336754  2013      9   30    2147.0            2137       10.0      30.0   
336755  2013      9   30    2149.0            2156       -7.0    2245.0   
336756  2013      9   30    2150.0            2159       -9.0    2250.0   
336757  2013      9   30    2159.0            1845      194.0    2344.0   
336758  2013      9   30    2203.0            2205       -2.0    2339.0   
336759  2013      9   30    2207.0            2140       27.0    2257.0   
336760  2013      9   30    2211.0            2059       72.0    2339.0   
336761  2013      9   30    2231.

create a new dataframe with only four columns by dropping all the rest of the columns. The table show the averge speed of the carriers flying from the 3 airports to each destinations.

In [192]:
#yy.groupby('month').size().describe()
yy=speed_df
yy=yy.drop(['year','month','day','dep_time','sched_dep_time','dep_delay','arr_time','sched_arr_time','arr_delay','flight','tailnum','air_time','distance','hour','minute','time_hour'],axis=1)
yy.groupby(['origin','carrier','dest']).mean().unstack().reset_index()

origin carrier     speed                                         \
dest                      ABQ       ACK      ALB       ANC       ATL   
0       EWR      9E       NaN       NaN      NaN       NaN  6.731659   
1       EWR      AA       NaN       NaN      NaN       NaN       NaN   
2       EWR      AS       NaN       NaN      NaN       NaN       NaN   
3       EWR      B6       NaN       NaN      NaN       NaN       NaN   
4       EWR      DL       NaN       NaN      NaN       NaN  6.775702   
5       EWR      EV       NaN       NaN  4.53835       NaN  6.599092   
6       EWR      MQ       NaN       NaN      NaN       NaN       NaN   
7       EWR      OO       NaN       NaN      NaN       NaN       NaN   
8       EWR      UA       NaN       NaN      NaN  8.166432  6.609620   
9       EWR      US       NaN       NaN      NaN       NaN       NaN   
10      EWR      VX       NaN       NaN      NaN       NaN       NaN   
11      EWR      WN       NaN       NaN      NaN       NaN       NaN   
12      JFK      9E       NaN       NaN      NaN       NaN  6.331007   
13      JFK      AA       NaN       NaN      NaN       NaN       NaN   
14      JFK      B6  7.370106  4.815894      NaN       NaN       NaN   
15      JFK      DL       NaN       NaN      NaN       NaN  6.848041   
16      JFK      EV       NaN       NaN      NaN       NaN  6.440678   
17      JFK      HA       NaN       NaN      NaN       NaN       NaN   
18      JFK      MQ       NaN       NaN      NaN       NaN       NaN   
19      JFK      UA       NaN       NaN      NaN       NaN       NaN   
20      JFK      US       NaN       NaN      NaN       NaN       NaN   
21      JFK      VX       NaN       NaN      NaN       NaN       NaN   
22      LGA      9E       NaN       NaN      NaN       NaN       NaN   
23      LGA      AA       NaN       NaN      NaN       NaN       NaN   
24      LGA      B6       NaN       NaN      NaN       NaN       NaN   
25      LGA      DL       NaN       NaN      NaN       NaN  6.786814   
26      LGA      EV       NaN       NaN      NaN       NaN  6.457627   
27      LGA      F9       NaN       NaN      NaN       NaN       NaN   
28      LGA      FL       NaN       NaN      NaN       NaN  6.705755   
29      LGA      MQ       NaN       NaN      NaN       NaN  6.751775   
30      LGA      OO       NaN       NaN      NaN       NaN       NaN   
31      LGA      UA       NaN       NaN      NaN       NaN       NaN   
32      LGA      US       NaN       NaN      NaN       NaN       NaN   
33      LGA      WN       NaN       NaN      NaN       NaN  6.230008   
34      LGA      YV       NaN       NaN      NaN       NaN       NaN   

                                      ...                                   \
dest       AUS       AVL       BDL    ...          SNA       SRQ       STL   
0          NaN       NaN       NaN    ...          NaN       NaN       NaN   
1          NaN       NaN       NaN    ...          NaN       NaN       NaN   
2          NaN       NaN       NaN    ...          NaN       NaN       NaN   
3          NaN       NaN       NaN    ...          NaN       NaN       NaN   
4          NaN       NaN       NaN    ...          NaN       NaN       NaN   
5          NaN  6.535034  4.618715    ...          NaN       NaN  6.486219   
6          NaN       NaN       NaN    ...          NaN       NaN       NaN   
7          NaN       NaN       NaN    ...          NaN       NaN       NaN   
8     7.170660       NaN  4.504233    ...     7.418502       NaN  7.207104   
9          NaN       NaN       NaN    ...          NaN       NaN       NaN   
10         NaN       NaN       NaN    ...          NaN       NaN       NaN   
11    7.166129       NaN       NaN    ...          NaN       NaN  6.676250   
12    6.852603       NaN       NaN    ...          NaN       NaN       NaN   
13    7.117357       NaN       NaN    ...          NaN       NaN       NaN   
14    7.169164       NaN       NaN    ...          NaN  6.853914       NaN   
15    7.221568